tls/s2n_server_hello.c

```diff
@@ -105,6 +105,14 @@ static int s2n_server_hello_parse(struct s2n_connection *conn)
     GUARD(s2n_stuffer_read_bytes(in, protocol_version, S2N_TLS_PROTOCOL_VERSION_LEN));
     GUARD(s2n_stuffer_read_bytes(in, conn->secure.server_random, S2N_TLS_RANDOM_DATA_LEN));
 
+    /* If the client receives a second HelloRetryRequest in the same connection, it MUST send an error. */
+    if (s2n_is_hello_retry_valid(conn)) {
+        if (s2n_is_hello_retry_handshake(conn)) {
+            S2N_ERROR(S2N_ERR_BAD_MESSAGE);
+        }
```

Comment:
You could use ENSURE_POSIX instead. But is there a more specific error that's relevant?

Reply:
There isn't, I can add a new error code `S2N_ERR_INVALID_HELLO_RETRY_STATE`